controllers/argocd/keycloak.go

```diff
@@ -892,22 +940,48 @@ func createRealmConfig(cfg *keycloakConfig) ([]byte, error) {
 					},
 				},
 			},
+			{
+				Name:     "profile",
+				Protocol: "openid-connect",
+				Attributes: map[string]string{
+					"include.in.token.scope":    "true",
+					"display.on.consent.screen": "true",
+				},
+			},
 		},
 	}
 
 	// Add OpenShift-v4 as Identity Provider only for OpenShift environment.
 	// No Identity Provider is configured by default for non-openshift environments.
 	if IsTemplateAPIAvailable() {
+		baseURL := "https://kubernetes.default.svc.cluster.local"
+		if isProxyCluster() {
+			baseURL = getOpenShiftAPIURL()
+		}
+
 		ks.IdentityProviders = []*keycloakv1alpha1.KeycloakIdentityProvider{
 			{
 				Alias:       "openshift-v4",
 				DisplayName: "Login with OpenShift",
 				ProviderID:  "openshift-v4",
 				Config: map[string]string{
-					"baseUrl":      "https://kubernetes.default.svc.cluster.local",
+					"baseUrl":      baseURL,
 					"clientSecret": oAuthClientSecret,
```

Comment:
All these hardcoded secrets look really insecure. Can they be pulled from somewhere more secure?

Reply:
can you be more specific on which one ? the one that you pointed is just the service name that point to cluster.